samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -436,6 +439,58 @@ static void update_cpi(u16_t cpi)
 	}
 }
 
+static void update_downshift_time(u8_t reg_addr, u32_t time)
+{
+	/* Set downshift time:
+	 * - Run downshift time (from Run to Rest1 mode)
+	 * - Rest 1 downshift time (from Rest1 to Rest2 mode)
+	 * - Rest 2 downshift time (from Rest2 to Rest3 mode)
+	 */
+	u32_t maxtime;
+	u32_t mintime;
+
+	if (reg_addr == OPTICAL_REG_RUN_DOWNSHIFT) {
+		/*
+		 * Run downshift time = OPTICAL_REG_RUN_DOWNSHIFT * 10 ms
+		 */
+		maxtime = 2550;
+		mintime = 10;
+	} else if (reg_addr == OPTICAL_REG_REST1_DOWNSHIFT) {
+		/*
+		 * Rest1 downshift time = OPTICAL_REG_RUN_DOWNSHIFT
+		 *                        * 320 * Rest1 rate (default 1 ms)
+		 */
+		maxtime = 81600;
+		mintime = 320;
+	} else if (reg_addr == OPTICAL_REG_REST2_DOWNSHIFT) {
+		/*
+		 * Rest2 downshift time = OPTICAL_REG_REST2_DOWNSHIFT
+		 *                        * 32 * Rest2 rate (default 100 ms)
+		 */
+		maxtime = 816000;
+		mintime = 3200;
+	} else {
+		LOG_ERR("Not supported");
+		return;
+	}
+
+	if ((time > maxtime) || (time < mintime)) {
+		LOG_WRN("Downshift time %u out of range", time);
+		return;
+	}
+
+	/* Convert time to register value */
+	u8_t value = time / mintime;
```

Comment:
value will never be zero? is zero not a correct option?

Reply:
Zero is not a correct option, at least for this sensor. Added assert, I believe assert check during development is enough